pkg/cluster/samples.go

```diff
@@ -19,17 +20,22 @@ func (m *manager) disableSamples(ctx context.Context) error {
 		return nil
 	}
 
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		c, err := m.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
+	return retry.OnError(
+		retry.DefaultRetry,
+		func(err error) bool {
+			return errors.IsConflict(err) || errors.IsNotFound(err)
+		},
+		func() error {
+			c, err := m.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
 
-		c.Spec.ManagementState = operatorv1.Removed
+			c.Spec.ManagementState = operatorv1.Removed
```

Comment:
LGTM, a minor nit, GitHub is highlighting whitespace here, maybe this should be a tab space?